### ISSUE TITLE
Frame lags when filling up larger holes

### ISSUE THREAD

**Issue by ElerWohlmuthFH** [pmdevita/CreeperHeal2#8]
Hey,
First of all, great plugin. Unfortunately I have the problem, that when larger holes are filling up (charged creeper sized), my fps (client side) go down from 250 to about 15-40fps. I have no clue why, since it kinda does not make any sense. Do you have an idea why it could be?
cheers

**Comment by pmdevita** [comment]
Can you paste your config.yml here? Thanks!

**Comment by ElerWohlmuthFH** [comment]
`# CreeperHeal2

# Delay before a hole is fixed in seconds
initial-delay: 15

# Delay between individual block replacements in ticks
# Zero makes it replace everything at once
between-blocks-delay: 10

# Types of explosions to handle
types:
  creeper: true
  tnt: false
  ghast: false
  wither: false
  minecart-tnt: false
  ender-crystal: false
  bed: false

# Turbo Mode
# Turbo Mode can help repair larger explosions faster.
# The minimum number of blocks an explosion should involve before turbo mode is activated
# 0 disables turbo
turbo-threshold: 0

# Turbo type lets you determine if turbo replaces at a constant
# amount or a percentage of the remaining total
# 0 - constant amount
# 1 - percentage
turbo-type: 1

# Constant Amount Turbo Settings
# The number of blocks to replace in each replacement cycle (For reference, non-turbo is 1)
# Raising this will slightly increase visible lag. Recommended <20
turbo-amount: 3

# Percentage Turbo Settings
# Percentage of blocks to replace each cycle (1-100). Recommended <20
turbo-percentage: 1

# Highest amount of blocks allowed to be replaced in one cycle
# Important to set if you increase turbo-percentage to prevent massive
# lag-spiking replacement cycles. Recommended <20
turbo-cap: 10

# Allow the chaining of TNT explosions rather than just saving the TNT block
explode-tnt: true

# Enable stats/usage data collection
bstats: true

# Enable debug mode
debug: false`

**Comment by pmdevita** [comment]
Hmm that is very strange, this really shouldn't be caused by the plugin. Is there anything else that might cause this? Do smaller explosions healing result in FPS drop? Try raising `between-blocks-delay` to 20 and then try 80, does this help reduce the drop?

**Comment by ElerWohlmuthFH** [comment]
With smaller explosions I do not notice it that much, if I raise the between-blocks-delay to 20 there is no lag. But it's kinda weird, since i seem to be the only one to have that lag (i am NOT hosting on my PC). Also, I feel like there is no difference in between-blocks-delay between anything from 10-19 and then with 20 ticks its suddently a lot slower. I tried running it on a spigot server without any other plugins, but no difference.

**Comment by pmdevita** [comment]
OK, it sounds like you are the only one experiencing the lag, is that correct? That likely narrows it down to a problem with your computer. What kind of internet and network connection do you have? Maybe try uninstalling and reinstalling Minecraft? What are your specs?

**Comment by ElerWohlmuthFH** [comment]
Yes I am. When I join on my Notebook, I also do NOT have that lag. Only on my PC, which should be more than powerful enough:

- NVIDIA GTX 970
- i7 4790k (4 Cores, overclocked to 4,6 GHZ)
- 32 GB of RAM, of which I reserved 16 for Minecraft.


I already tried it with a new, clean Installation of Minecraft, and with Vanilla and Opfitine and the problem still remains. I am connected by Ethernet cable and have a really stable connection with plenty of download and upload speed. Ping is only a few ms since the server is running on my other PC. Also, its weird since those are FPS lags, it just does not make any sense. I just wanted to let you know of the problem because you might have known a simple fix. If I find a fix myself, I will let you know.

**Comment by pmdevita** [comment]
Yeah that definitely doesn't seem like a specs problem. Maybe it has something to do with block updates? Try going into a creative world on the server with some other players and have them spam blocks around and see if it lags then. Or use some other plugin that builds stuff if you can't get other players on. Does the lag happen if you are further away from the healing or not in range to load the chunk it's happening in?
